hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecord.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.common.model;
+
+public class HoodieAvroRecord<T extends HoodieRecordPayload> extends HoodieRecord<T> {
```

Comment:
Should be `HoodieRecordPayload<T>`

Reply:
Right.  This is more like a retrofit for Java 7 style and users extending old HoodieRecord.  This should be fixed in one shot with new APIs.

Reply:
@yihua sorry, not sure i understood your point. Can you elaborate?

Why do we want to extend `HoodieAvroRecord` with format-specific impl?

Reply:
Discussed offline: this will be an intermediate state until RFC-46 is fully implemented

Reply:
As discussed, this is more of an intermediate solution for row writer, before RFC-46 revamps it completely.